### PR TITLE
First stages of Peacock improvements

### PR DESCRIPTION
**Pull request by aeslaughter**

### CODE REVIEW

File: python/peacock/ExodusViewer/ExodusPluginManager.py [hunk 1]
```diff
@@ -18,43 +18,18 @@ class ExodusPluginManager(QtWidgets.QWidget, peacock.base.PluginManager):
...
-        self.LeftScrollArea.viewport().setStyleSheet(".QWidget { background: transparent; }");
-        self.LeftScrollArea.setFrameShape(QtWidgets.QFrame.NoFrame)
-        self.LeftScrollArea.setHorizontalScrollBarPolicy(QtCore.Qt.ScrollBarAlwaysOff)
-        self.LeftScrollContent = QtWidgets.QWidget()
-        self.LeftScrollArea.setWidget(self.LeftScrollContent)
+        self.MainLayout = QtWidgets.QHBoxLayout(self)
         self.LeftLayout = QtWidgets.QVBoxLayout()
-        self.LeftScrollContent.setLayout(self.LeftLayout)
-        self.LeftLayout.setContentsMargins(0, 0, 0, 0)
+        #self.LeftLayout.setSpacing(10)
```
Comment (brianmoose) [inline python/peacock/ExodusViewer/ExodusPluginManager.py]: Remove?

File: python/peacock/ExodusViewer/plugins/BlockPlugin.py [hunk 2]
```diff
@@ -10,153 +10,206 @@
...
+    def onWindowUpdated(self):
+        """
+        Update the blocks if needed.
         """
-        Updates the visible block/nodesets/sidesets based on the selector widget settings.
+        blocks = self._blocksChanged(self.BlockSelector, chigger.exodus.ExodusReader.BLOCK)
+        sideset = self._blocksChanged(self.SidesetSelector, chigger.exodus.ExodusReader.SIDESET)
+        nodeset = self._blocksChanged(self.NodesetSelector, chigger.exodus.ExodusReader.NODESET)
+        if any([blocks, sideset, nodeset]):
+            print 'reset...', [blocks, sideset, nodeset]
```
Comment (brianmoose) [inline python/peacock/ExodusViewer/plugins/BlockPlugin.py]: Should this be a print? Or a `mooseMessage`?

File: python/peacock/ExodusViewer/plugins/MeshPlugin.py [hunk 3]
```diff
@@ -239,15 +376,17 @@ def main(size=None):
...
     return widget, widget.VTKWindowPlugin
 
 if __name__ == '__main__':
     from peacock.utils import Testing
     app = QtWidgets.QApplication(sys.argv)
-    widget, window = main()
-    filename = Testing.get_chigger_input('mug_blocks_out.e')
-    window.initialize([filename])
+    filenames = Testing.get_chigger_input_list('mug_blocks_out.e', 'displace.e')
+    #filenames = Testing.get_chigger_input_list('diffusion_1.e', 'diffusion_2.e')
```
Comment (brianmoose) [inline python/peacock/ExodusViewer/plugins/MeshPlugin.py]: remove?

**Review by brianmoose — CHANGES_REQUESTED** [review]
Have you tried this out on linux?
Once all the tests are working I will give it a spin on both mac and linux.
Somebody who actually uses peacock will probably also want to try it out.

File: python/peacock/ExodusViewer/plugins/MeshPlugin.py (hunk 3 above)
Comment (aeslaughter) [reply python/peacock/ExodusViewer/plugins/MeshPlugin.py]: I would like to keep this there, this is for debugging.

**Review by brianmoose — CHANGES_REQUESTED** [review]
Toggling the mesh block no longer turns off the mesh viewer.

The color bar sometimes seems to show up in the mesh viewer. To reproduce, open simple_diffusion.i, change the mesh `dim` to 20, apply, then change it back to 2 and apply.

**Review by brianmoose — CHANGES_REQUESTED** [review]
On linux, just trying to run `simple_diffusion.i` results in a crash with
```
Traceback (most recent call last):
  File "/opt/civet/build_0/moose/python/peacock/ExodusViewer/plugins/GoldDiffPlugin.py", line 131, in onSetVariable
    self.updateOptions()
  File "/opt/civet/build_0/moose/python/peacock/ExodusViewer/plugins/GoldDiffPlugin.py", line 201, in updateOptions
    elif (not diff) and self.DiffVTKWindow.isVisible():
AttributeError: 'GoldDiffPlugin' object has no attribute 'DiffVTKWindow'
```
followed by a core dump

**Review by brianmoose — DISMISSED** [review]
Still getting the same problem with the crash on linux.

Mesh stuff on the input tab seems to be working now.

So we no longer have the gold diff viewer? Or does that only show up when it is an option?

It looks intentional but you can no longer set a custom background color,  just gradient, black, or white.

**Comment by aeslaughter** [comment]
> Still getting the same problem with the crash on linux.

I will need to fire this up on a linux machine I suppose.

> Mesh stuff on the input tab seems to be working now.

Good.

> So we no longer have the gold diff viewer? Or does that only show up when it is an option?

It only appears when a gold file exists.

> It looks intentional but you can no longer set a custom background color, just gradient, black, or white.

You can still change the gradient colors via the preferences.

**Comment by permcody** [comment]
I love the name of that last commit. So does that commit "mess up something" or "reverse the previously messed up something"? 😆

**Comment by aeslaughter** [comment]
@permcody That was me just pushing WIP commits w/o remembering that I had an open PR.

**Comment by aeslaughter** [comment]
@brianmoose Just kidding about being ready, I found one more small thing I need to fix up.

**Review by brianmoose — CHANGES_REQUESTED** [review]
Same crash on linux.

The "view mesh" preference doesn't seem to be honored in the ExodusViewer.

Not sure how you are intending the preferences to work.
Before they were just used as default values and there was always the option to override it (via separate widgets). 

Now if I want my hot pink background I have to set it in the preferences and restart peacock. Even if you auto update things when the preferences change, it might still be nice to be able to override it just for the current session.

File: python/peacock/ExodusViewer/plugins/GoldDiffPlugin.py [hunk 4]
```diff
@@ -89,25 +80,30 @@ def __init__(self, size=None):
...
 
         self.GoldVTKWindow = ExternalVTKWindowPlugin(self.GoldToggle, size=size, text='GOLD')
-        self.DiffVTKWindow = ExternalVTKWindowPlugin(self.DiffToggle, size=size, text='EXODIFF')
-
-        self.setup()
-        self.setCollapsed(True)
 
         # Locate MOOSE exodiff program
         self._exodiff = None
         moose_dir = os.path.abspath(os.path.join(os.path.realpath(inspect.getfile(self.__class__)), '..', '..', '..'))
```
Comment (brianmoose) [inline python/peacock/ExodusViewer/plugins/GoldDiffPlugin.py]: This is what was causing the crash on linux. It is actually reproducible on mac as well. You just need
to `unset MOOSE_DIR` then 
`python/peacock/peacock -e modules/combined/combined-devel -i python/peacock/tests/common/simple_diffusion.i --run`

Adding  `"..", ".."` to the statement seems to fix the immediate issue but the case where exodiff is not found will still break.

Comment (aeslaughter) [reply python/peacock/ExodusViewer/plugins/GoldDiffPlugin.py]: Thanks for finding this, I will figure it out next week or you can just push on top of this if you want to take care of it.

**Comment by aeslaughter** [comment]
> Now if I want my hot pink background I have to set it in the preferences and restart peacock. Even if you auto update things when the preferences change, it might still be nice to be able to override it just for the current session.

You should not have to restart peacock, the background preferences should take immediate affect. I  will add a control for the current session only.

**Review by brianmoose — CHANGES_REQUESTED** [review]
Having a problem with the background gradient.
To reproduce:
Start peacock then Peacock->Preferences->ExodusViewer and uncheck "Use background gradient"
Exit peacock and start it up again. Run an input file and look at the results in ExodusViewer. The background now uses a solid color, as expected.  
Clicking on the Results->Background->Gradient menu item doesn't turn on the gradient background.

**Review by brianmoose — DISMISSED** [review]
Looks OK to me.
Somebody who actually runs peacock might want to take a look before merging.

**Review by friedmud — DISMISSED** [review]
The block_kernel_test.i in moose/test/tests/kernels/block_kernel is still showing "double updating" of the Exodus solution when moving through timesteps

**Review by brianmoose — APPROVED** [review]

**Pull request merged**